dev_scripts/heat_tables.py

```diff
@@ -0,0 +1,134 @@
+import matplotlib.pyplot as plt
+from utils.files import gather_result_files
+from utils.constants import GOODAI_RED, GOODAI_GREEN
+from reporting.results import TestResult
+from collections import defaultdict
+
+
+LIGHT_GREY = (0.424, 0.447, 0.459)
+DARK_GREY = (0.302, 0.318, 0.325)
+RUN_NAMES = [
+    "Benchmark 3 - 32k (isolated)",
+    "Benchmark 3 - 1k",
+    "Benchmark 3 - 32k",
+    "Benchmark 3 - 120k",
+    "Benchmark 3 - 200k",
+    "Benchmark 3 - 500k",
+]
+TO_RETRIEVE = {
+    span: [
+        "LLMChatSession - together_ai-mistralai-Mixtral-8x7B-Instruct-v0.1 - 32768",
+        "LLMChatSession - together_ai-mistralai-Mixtral-8x22B-Instruct-v0.1 - 65536",
+        "LLMChatSession - together_ai-meta-llama-Llama-3-70b-chat-hf - 8000",
+        "LLMChatSession - gpt-3.5-turbo - 16384",
+        "LLMChatSession - gpt-4-turbo-2024-04-09 - 128000",
+        "LLMChatSession - gpt-4o - 128000",
+        "LLMChatSession - claude-3-opus - 200000",
+        "GeminiProInterface",
+        "LTMAgentWrapper - together_ai-meta-llama-Llama-3-70b-chat-hf - 8000 - QG_JSON_USER_INFO",
+        "LTMAgentWrapper - gpt-4-turbo - 16384 - QG_JSON_USER_INFO",
+        "LTMAgentWrapper - claude-3-opus - 16384 - QG_JSON_USER_INFO",
+    ] for span in RUN_NAMES
+}
+ALIASES = {
+    "LLMChatSession - together_ai-mistralai-Mixtral-8x7B-Instruct-v0.1 - 32768": "Mixtral 8x7B",
+    "LLMChatSession - together_ai-mistralai-Mixtral-8x22B-Instruct-v0.1 - 65536": "Mixtral 8x22B",
+    "LLMChatSession - together_ai-meta-llama-Llama-3-70b-chat-hf - 8000": "Llama 3 70B",
+    "LLMChatSession - gpt-3.5-turbo - 16384": "GPT-3.5 turbo",
+    "LLMChatSession - gpt-4-turbo-2024-04-09 - 128000": "GPT-4 turbo",
+    "LLMChatSession - gpt-4o - 128000": "GPT-4o",
+    "LLMChatSession - claude-3-opus - 200000": "Claude 3 Opus",
+    "GeminiProInterface": "Gemini 1.5 Pro",
+    "LTMAgentWrapper - together_ai-meta-llama-Llama-3-70b-chat-hf - 8000 - QG_JSON_USER_INFO": "LTM (Llama 3 70B)",
+    "LTMAgentWrapper - gpt-4-turbo - 16384 - QG_JSON_USER_INFO": "LTM (GPT-4 turbo)",
+    "LTMAgentWrapper - claude-3-opus - 16384 - QG_JSON_USER_INFO": "LTM (Claude 3 Opus)",
+}
+
+
+# Exceptions
+def set_alt_name(idx: int, mem_spans: list[str], alt_name: str):
+    ALIASES[alt_name] = ALIASES[TO_RETRIEVE["Benchmark 3 - 1k"][idx]]
+    for span in mem_spans:
+        TO_RETRIEVE[f"Benchmark 3 - {span}"][idx] = alt_name
+
+
+set_alt_name(0, ["1k", "32k"], "LLMChatSession - together_ai-mistralai-Mixtral-8x7B-Instruct-v0.1 - 32000")
+set_alt_name(4, ["120k", "200k"], "LLMChatSession - gpt-4-turbo - 128000")
+set_alt_name(10, ["120k", "200k", "500k"], "LTMAgentWrapper - claude-3-opus-20240229 - 16384 - QG_JSON_USER_INFO")
+set_alt_name(9, ["200k", "500k"], "LTMAgentWrapper - gpt-4-turbo-2024-04-09 - 16384 - QG_JSON_USER_INFO")
```

Comment:
I have no idea what these function calls are doing

Reply:
Ohhh, I think I see. The context sizes are a little different for these agents in some runs, so this is replacing those names under those runs.

Reply:
With those calls I'm basically saying _"The agent at index 0 will take this other name for spans 1k and 32k"_. I needed it because the results were sometimes saved like `claude-3-opus` and other times as `claude-3-opus-20240229`.